src/felice/admin.clj

```diff
@@ -0,0 +1,367 @@
+(ns felice.admin
+  (:require [clojure.walk :refer [stringify-keys]]
+            [clojure.spec.alpha :as s]
+            [clojure.string :as str])
+  (:import org.apache.kafka.clients.admin.AdminClient
+           org.apache.kafka.clients.admin.NewTopic
+           org.apache.kafka.common.config.TopicConfig
+           org.apache.kafka.common.Node
+           org.apache.kafka.common.TopicPartition
+           org.apache.kafka.common.TopicPartitionInfo
+           org.apache.kafka.clients.consumer.OffsetAndMetadata))
+
+(defn- ->node
+  [^Node node]
+  (cond-> {:host (.host node)
+           :port (.port node)
+           :id (.id node)}
+    (.hasRack node) (assoc :rack (.rack node))))
+
+(defn- ->topic-partition
+  [^TopicPartitionInfo partition]
+  {:isr (map ->node (.isr partition))
+   :leader (->node (.leader partition))
+   :replicas (map ->node (.replicas partition))})
+
+(defn- ->offset-metadata
+  [^OffsetAndMetadata offset-metadata]
+  {:metadata (.metadata offset-metadata)
+   :offset (.offset offset-metadata)})
+
+
+(defn admin-client
+  "Instanciate an `AdminClient` from properties"
+  {:added "3.2.0-1.7"}
+  ^AdminClient
```

Comment:
We should code against the [Admin](https://kafka.apache.org/26/javadoc/org/apache/kafka/clients/admin/Admin.html) interface instead of AdminClient

From AdminClient javadoc
> The base class for in-built admin clients. Client code should use the newer [Admin](https://kafka.apache.org/26/javadoc/org/apache/kafka/clients/admin/Admin.html) interface in preference to this class. This class may be removed in a later release, but has not be marked as deprecated to avoid unnecessary noise.